src/main/groovy/io/openliberty/tools/gradle/tasks/DevTask.groovy

```diff
@@ -1262,15 +1264,19 @@ class DevTask extends AbstractFeatureTask {
 
         File buildFile = project.getBuildFile();
 
+        List<Path> webResourceDirs = LooseWarApplication.getWebSourceDirectoriesToMonitor(project);
         // Instantiate util before any child gradle tasks launched so it can help find available port if needed
+        List<ProjectModule> projectModules = getProjectModules()
+        allProjects.add(project)
+        allProjects.addAll(DevTaskHelper.getAllUpstreamProjects(project))
```

Comment:
The call to `getProjectModules` seems to be doing this exact same thing...why? Can we pass in a collection to that method and let it populate `allProjects` for us?

Reply:
allProjects contains projectmodules plus the ear project

Reply:
I have removed those. Please check now..
We need list of all dependent projects to construct the projectmodule object
We need list of all dependent project+ current ear project in allProjects variable, which is not used now, its going to be used in updateArtifacts in further PRs